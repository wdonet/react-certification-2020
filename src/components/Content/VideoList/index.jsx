import React, { useContext } from 'react';
import Video from '../Video';
import YoutubeContext from '../../../context/YoutubeContext.jsx';

const VideoList = () => {
  const listofvideos = useContext(YoutubeContext).slice(1);

  const videos = listofvideos.map((video) => (
    <Video key={video.id.videoId} info={video} />
  ));

  return (
    <table className="list-videos">
      <tbody>{videos}</tbody>
    </table>
  );
};

export default VideoList;
