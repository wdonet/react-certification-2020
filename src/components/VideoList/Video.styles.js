import styled from 'styled-components';

const Container = styled.div``;

const TitleBar = styled.div`
  padding: 15px;
  background-color: rgba(225,6,0, 0.8);
  color: #ffffff;
`;

const Title = styled.h2`
  font-weight: 100;
`;

const List = styled.div`
  padding: 0 15px;
`;

const Styled = { Container, TitleBar, Title, List };
export default Styled;
