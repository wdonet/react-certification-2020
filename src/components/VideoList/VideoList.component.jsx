import React from 'react';
import Styled from './Video.styles';
import Card from '../Card';

const VideoList = ({ items }) => {
  return (
    <Styled.Container>
      <Styled.TitleBar>
        <Styled.Title>Wizeline's Youtube Videos</Styled.Title>
      </Styled.TitleBar>
      <Styled.List>
        {items.map(({ snippet }) => (
          <Card
            title={snippet.title}
            thumbnail={snippet.thumbnails.default.url}
            description={snippet.description || "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum feugiat nulla lectus. Maecenas at sapien eu nisi hendrerit eleifend. Duis quis condimentum ligula..."}
          />
        ))}
      </Styled.List>
    </Styled.Container>
  );
};

export default VideoList;
