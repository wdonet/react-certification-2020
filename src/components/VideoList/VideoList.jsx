import React from 'react';
import styled from 'styled-components';
import VideoCard from '../VideoCard';

const VideoListContainer = styled.div`
  text-align: center;
`;

export default function VideoList({ videos, handleVideoSelected }) {
  if (!videos) {
    return <div />;
  }
  const renderedVideos = videos.map((video) => {
    return (
      <VideoCard
        key={Math.random()}
        video={video}
        handleVideoSelected={handleVideoSelected}
      />
    );
  });
  return <VideoListContainer>{renderedVideos}</VideoListContainer>;
}
