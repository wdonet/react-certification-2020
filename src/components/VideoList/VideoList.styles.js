import styled from 'styled-components';

export const Container = styled.div``;

export const Title = styled.h1`
  text-align: center;
`;

export const List = styled.div`
  padding: 0 15px;
  display: flex;
  flex-flow: row wrap;
`;
