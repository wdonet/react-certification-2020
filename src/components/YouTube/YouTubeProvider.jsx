import React, { useContext, useReducer } from 'react';

const YoutTubeContext = React.createContext();

function youtubeReducer(state, action) {
  switch (action.type) {
    case 'search':
      return {
        ...state,
        search: action.payload,
      };
    case 'currentVideo':
      return {
        ...state,
        currentVideo: action.payload,
      };
    case 'closeCurrentVideo':
      return {
        ...state,
        currentVideo: null,
      };
    case 'switchTheme':
      return {
        ...state,
        theme: state.theme === 'dark' ? 'default' : 'dark',
      };
    default: {
      return state;
    }
  }
}

function YouTubeProvider({ children }) {
  const [state, dispatch] = useReducer(youtubeReducer, {});
  return (
    <YoutTubeContext.Provider value={{ state, dispatch }}>
      {children}
    </YoutTubeContext.Provider>
  );
}

function useYouTube() {
  const context = useContext(YoutTubeContext);
  if (context === undefined) {
    throw new Error('useYouTube must be used within YouTubeProvider');
  }
  return context;
}

export { YouTubeProvider, useYouTube };
