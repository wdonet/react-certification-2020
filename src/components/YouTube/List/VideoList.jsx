import React from 'react';
import useVideos from '../useVideos';

import {
  VideosContainer,
  VideoCard,
  VideoPreview,
  VideoDescription,
  VideoTitle,
  VideoContent,
} from './VideoList.styled';
import { useYouTube } from '../YouTubeProvider';

const VideoList = () => {
  const { state, dispatch } = useYouTube();
  const { search } = state;

  const { videos, isLoading, error } = useVideos({ search });

  const onVideoCardClick = (item) => {
    dispatch({ type: 'currentVideo', payload: item });
  };

  if (isLoading) return <p>Loading...</p>;

  return (
    <>
      {error && (
        <p style={{ textAlign: 'center' }}>
          Error from YouTube API, displaying mock data...
        </p>
      )}
      <VideosContainer>
        {videos.map((item) => (
          <VideoCard key={item.id.videoId} onClick={() => onVideoCardClick(item)}>
            <VideoPreview src={item.snippet.thumbnails.medium.url} />
            <VideoContent>
              <VideoTitle>{item.snippet.title}</VideoTitle>
              <VideoDescription>{item.snippet.description}</VideoDescription>
            </VideoContent>
          </VideoCard>
        ))}
      </VideosContainer>
    </>
  );
};

export default VideoList;
