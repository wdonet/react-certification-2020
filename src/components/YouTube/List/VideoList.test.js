/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';

import VideoList from './VideoList';
import mockData from './youtube-videos-mock.json';
import { YouTubeProvider } from '../YouTubeProvider';

describe('VideoList', () => {
  let props;
  beforeEach(() => {
    props = {
      videos: mockData.items,
    };
  });
  it('should render VideoList component correctly', () => {
    const wrapper = shallow(
      <YouTubeProvider>
        <VideoList {...props} />
      </YouTubeProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
