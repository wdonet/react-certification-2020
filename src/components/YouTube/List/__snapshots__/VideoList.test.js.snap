// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoList should render VideoList component correctly 1`] = `ShallowWrapper {}`;
